commons/src/main/java/org/infinispan/commons/hash/MurmurHash3.java

```diff
@@ -388,8 +388,8 @@ public int hash(int hashcode) {
    public int hash(Object o) {
       if (o instanceof byte[])
          return hash((byte[]) o);
-      else if (o instanceof WrappedByteArray) {
-         return hash(((WrappedByteArray) o).getBytes());
+      else if (o instanceof WrappedBytes) {
+         return hash(((WrappedBytes) o).getBytes());
```

Comment:
shouldn't hash only the array from `WrappedBytes.backArrayOffset()` to `WrappedBytes.getLength()`?
the javadoc for `WrappedBytes.getBytes()` says it can return `null`. I think it must check for `null` before invoking the `hash()`.

Reply:
No, implementation currently does this. There are places all over that assume this currently, so I didn't want to complicate this PR.

Reply:
ours `WrappedBytes` implementations? 
what if the user puts its own?

Reply:
Everything will break :) We don't support other implementations.